lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -554,8 +554,8 @@ protected function _addForeignKeyConstraint(ForeignKeyConstraint $constraint)
 
         $this->_fkConstraints[$name] = $constraint;
 
-        // Add an explicit index on the foreign key columns.
-        // If there is already an index that fulfils this requirements drop the request.
+        // Add an implicit index on the foreign key columns.
```

Comment:
This change is arguable: the DBAL is the user of the DB platform, so the DBAL creating the index makes it explicit. Although, it is implicit for the user because they didn't ask it to be created.

Reply:
If you look below, the index is stored in a property called `implicitIndexes`: https://github.com/doctrine/dbal/blob/f97ee9445186e2f5edd519579405df53521cdae4/lib/Doctrine/DBAL/Schema/Table.php#L562-L577

I'm not sure I'm making the right call here, but once thing I am sure of, is that the current naming is quite confusing… maybe we should avoid this terminology entirely, and use
- user-defined indexes
- DBAL-defined  indexes
- RDBMS-defined indexes

That comment would become:

> Add a DBAL-defined index on the foreign key columns. If there is already a user-defined or RDBMS-defined (really not sure here if such the RDBMS-defined indexes will appear in the introspection) that fulfills this requirement drop the request (that word request feels weird here). In the case of __construct calling this method during hydration from schema-details all these DBAL-defined indexes lead to duplicates (if the table was created with the DBAL, and because in that case the DBAL finds the same indexes it created implicitly the previous time). This creates computation overhead in this case, however no duplicate indexes are ever added (based on columns).

Reply:
I like the more specific naming. Should we keep implicit indexes named implicit? This way, it will match the existing terminology and still prevent the ambiguity between the user-defined and the DBAL-defined ones.

Reply:
I think the RDBMS-defined index is the one kind of index that will not appear in that comment, will push a reworked version of that comment in a few seconds.